rust/schema/annotations.py

```diff
@@ -1520,3 +1520,21 @@ class _:
     todo!()
     ```
     """
+
+class Callable(AstNode):
+    """
+    A callable. Either a `Function` or a `ClosureExpr`.
+    """
+    param_list: optional["ParamList"] | child
+    attrs: list["Attr"] | child
+
```

Comment:
you can add `body: optional[Expr] | child` here and drop it on the two derived classes, and then reintroduce the more specific return type in the `FuncionImpl.qll` file with
```codeql
override BlockExpr getBody() { result = Generated::Function.super.getBody() }
```
This technique might get problematic with `list` fields, as the generated `getA...` is `final` and its type cannot be overridden, but for `optional` the only generated auxiliary predicate is `has...` which remains valid.

Reply:
hmm, turns out that doesn't work out of the box, as the extractor would need an `into()` through an optional inserted into generated code, which is probably not worth the hassle. I have some ideas, but they're better off for a follow-up